File: baselibray/src/main/java/com/zmj/baselibray/BaseApplication.java
package com.zmj.baselibray;

import android.app.Application;

/**
 * Author : Zmj
 * Blog : https://blog.csdn.net/Zmj_Dns
 * GitHub : https://github.com/ZmjDns
 * Time : 2020/3/27
 * Description :
 */
public class BaseApplication extends Application {

    private static BaseApplication baseApplication;

    @Override
    public void onCreate() {
        super.onCreate();

        baseApplication = this;
    }

    public static BaseApplication getBaseApplication(){
        return baseApplication;
    }
}

File: baselibray/src/main/java/com/zmj/baselibray/BaseApplication.kt
package com.zmj.baselibray

import android.app.Application

/**
 * Author : Zmj
 * Blog : https://blog.csdn.net/Zmj_Dns
 * GitHub : https://github.com/ZmjDns
 * Time : 2020/4/6
 * Description :
 */
open class BaseApplication: Application() {

    companion object{
        public lateinit var baseApplication: Application
    }

    override fun onCreate() {
        super.onCreate()
        baseApplication = this
    }
}